### ISSUE TITLE
S3 fetch example is throwing AWS Credential error

### ISSUE THREAD

**Issue by rburhum** [locationtech-labs/geopyspark#183]
I followed the examples of pygeospark and the example is throwing errors when fetching S3 datasources:

```bash
(spark) rburhum@peru:/tmp$ python test.py
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
17/05/08 15:00:03 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    metadata = rdd.collect_metadata()
  File "/Users/rburhum/.virtualenvs/spark/lib/python3.6/site-packages/geopyspark-0.1.0-py3.6.egg/geopyspark/geotrellis/rdd.py", line 182, in collect_metadata
    json_metadata = self.srdd.collectMetadata(str(tile_size), crs)
  File "/usr/local/spark/python/lib/py4j-0.10.4-src.zip/py4j/java_gateway.py", line 1133, in __call__
  File "/usr/local/spark/python/lib/py4j-0.10.4-src.zip/py4j/protocol.py", line 319, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling o26.collectMetadata.
: com.amazonaws.SdkClientException: Unable to load AWS credentials from any provider in the chain
	at com.amazonaws.auth.AWSCredentialsProviderChain.getCredentials(AWSCredentialsProviderChain.java:131)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1110)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:759)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:723)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:716)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4194)
	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:4949)
	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:4923)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4178)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4141)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4135)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:838)
	at geotrellis.spark.io.s3.AmazonS3Client.listObjects(AmazonS3Client.scala:45)
	at geotrellis.spark.io.s3.S3Client$$anon$1.<init>(S3Client.scala:113)
	at geotrellis.spark.io.s3.S3Client$class.listObjectsIterator(S3Client.scala:112)
	at geotrellis.spark.io.s3.AmazonS3Client.listObjectsIterator(AmazonS3Client.scala:43)
	at geotrellis.spark.io.s3.S3InputFormat.getSplits(S3InputFormat.scala:127)
	at org.apache.spark.rdd.NewHadoopRDD.getPartitions(NewHadoopRDD.scala:125)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:252)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:250)
	at scala.Option.getOrElse(Option.scala:121)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:250)
	at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:252)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:250)
	at scala.Option.getOrElse(Option.scala:121)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:250)
	at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:252)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:250)
	at scala.Option.getOrElse(Option.scala:121)
	at org.apache.spark.rdd.RDD.partitions(RDD.scala:250)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1988)
	at org.apache.spark.rdd.RDD$$anonfun$reduce$1.apply(RDD.scala:1026)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:362)
	at org.apache.spark.rdd.RDD.reduce(RDD.scala:1008)
	at geotrellis.spark.TileLayerMetadata$.collectMetadataWithCRS(TileLayerMetadata.scala:147)
	at geotrellis.spark.TileLayerMetadata$.fromRdd(TileLayerMetadata.scala:237)
	at geotrellis.spark.package$withCollectMetadataMethods.collectMetadata(package.scala:193)
	at geopyspark.geotrellis.ProjectedRasterRDD.collectMetadata(RasterRDD.scala:179)
	at geopyspark.geotrellis.RasterRDD.collectMetadata(RasterRDD.scala:156)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:745)
```

I installed awscli and setup the authentication credentials and it still did not make a difference.

**Comment by jpolchlo** [comment]
Your AWS credentials are not available.  Are you running this from inside the docker container or locally?  In any event, you'll need to have a `.aws` directory created in your home directory with a `credentials` file and possibly a `config` file.  You may find this link useful: `http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html`.  If you are using this from the docker image, you'll need to include a volume mount with the command line argument `-v $HOME/.aws:/home/hadoop/.aws` to `docker run` after you set up your credentials.  We are currently chasing down a question of why credentials are required to access public S3 data, but for now, you'll need to do the steps above.

**Comment by jbouffard** [comment]
Resolved in #228
